internal/utils/validation.go

```diff
@@ -18,3 +18,53 @@ func StringIsDuration(i interface{}, k string) (warnings []string, errors []erro
 
 	return nil, nil
 }
+
+// StringIsElasticDuration is a SchemaValidateFunc which tests to make sure the supplied string is valid duration using Elastic time units:
+// d, h, m, s, ms, micros, nanos. (see https://www.elastic.co/guide/en/elasticsearch/reference/current/api-conventions.html#time-units)
+func StringIsElasticDuration(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		return nil, []error{fmt.Errorf("expected type of %s to be string", k)}
+	}
+
+	if v == "" {
+		return nil, []error{fmt.Errorf("%q contains an invalid duration: [empty]", k)}
+	}
+
+	firstPartCount := 0
+	for v != "" {
```

Comment:
I think this whole validation is clearer as a regex (untested, but something like`^(?:[0-9.])+(?:d|h|m|s|ms|micros|nanos)$`) WDYT?

Reply:
Clearly better with regex, I don't know what I was thinking 😕